src/datajudge/db_access.py

```diff
@@ -294,9 +294,12 @@ class DataReference:
     def __init__(
         self,
         data_source: DataSource,
-        columns: list[str] = None,
-        condition: Condition = None,
+        columns: Optional[list[str]] = None,
```

Comment:
@kklein are you aware if `columns` needs to be a list, aka, is the order of the elements relevant?

Reply:
I think we have some cases where we select the first element of a list (e.g. as a 'sample') with `columns[0]`. Besides this implementation detail I don't think we need ordering and I don't think we want repetition.

Reply:
E.g. here https://github.com/Quantco/datajudge/blob/main/src/datajudge/constraints/date.py#L171

Reply:
But that access is more like "hey, there should be only one".

Then, I'd suggest to consider changing the annotation to `Collection[str]` following the mantra 

> accept interfaces, return concrete types

Reply:
I'd much prefer that, too - I just wanted to chime in that some internal bits and pieces might need to be adapted; test would need to be adapted, too, to figure out which parts need adaptation).

Reply:
As a test, I randomly shuffled the Column Names in the constructor: only the Unique Constraints break.

The `get_uniques` in db_access.py constructs tuples whose order depends on the order in DataReference.
https://github.com/Quantco/datajudge/blob/main/src/datajudge/db_access.py#LL913C1-L913C1

The Uniques Between Constraints construct the list of tuples for both tables and then compare both lists. 
 For this, their DataReference must also have the same order, otherwise the tuples with values cannot be compared. Alternatively we could construct a dict

Reply:
Then, instead of `Collection` we for for `Sequence` for now.